airbyte-metrics/lib/src/main/java/io/airbyte/metrics/lib/DogStatsDMetricSingleton.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.metrics.lib;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.timgroup.statsd.NonBlockingStatsDClientBuilder;
+import com.timgroup.statsd.StatsDClient;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Light wrapper around the DogsStatsD client to make using the client slightly more ergonomic.
+ * <p>
+ * This class mainly exists to help Airbyte instrument/debug application on Airbyte Cloud.
+ * <p>
+ * Open source users are free to turn this on and consume the same metrics.
+ */
+@Slf4j
+public class DogStatsDMetricSingleton {
+
+  private static DogStatsDMetricSingleton instance;
+  private final StatsDClient statsDClient;
+  private final boolean instancePublish;
+
+  public static synchronized DogStatsDMetricSingleton getInstance() {
+    if (instance == null) {
+      throw new RuntimeException("You must initialize configuration with the initialize() method before getting an instance.");
+    }
+    return instance;
+  }
+
+  public synchronized static void initialize(final AirbyteApplication app, final boolean publish) {
+    initialize(app, publish, null, null);
+  }
+
+  @VisibleForTesting
+  public synchronized static void initialize(final AirbyteApplication app, final boolean publish, final String ddAgentHost, final String ddPort) {
+    if (instance != null) {
+      throw new RuntimeException("You cannot initialize configuration more than once.");
+    }
+
+    if (!publish) {
```

Comment:
it's a little hard for me to understand the different configurations.
1. line 43: if not publish then create the singleton but it won't publish anywhere
2. line 49: create singleton specifying dd host and port
3. line 55: ??? not sure.

it might be useful to have comments either in `initialize` or javadocs comments for the constructor to understand what each of these configurations mean.

Reply:
you are right, I've added comments and doc strings. Does this make more sense?

Reply:
some of this is invalidated by the env var change. please take another look

Reply:
nice!